gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.gameservices.samples.allocationpolicies;
+
+// [START cloud_game_servers_allocation_policy_create]
+
+import com.google.api.gax.longrunning.OperationSnapshot;
+import com.google.api.gax.retrying.RetryingFuture;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.cloud.gaming.v1alpha.AllocationPoliciesServiceClient;
+import com.google.cloud.gaming.v1alpha.AllocationPolicy;
+import com.google.cloud.gaming.v1alpha.CreateAllocationPolicyRequest;
+import com.google.protobuf.Any;
+import com.google.protobuf.Int32Value;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateAllocationPolicy {
+  public static void createAllocationPolicy(String projectId, String policyId) {
+    // String projectId = "your-project-id";
+    // String policyId = "your-policy-id";
+    try (AllocationPoliciesServiceClient client = AllocationPoliciesServiceClient.create()) {
+      String parent = String.format("projects/%s/locations/global", projectId);
+      String policyName = String.format("%s/allocationPolicies/%s", parent, policyId);
+
+      AllocationPolicy policy = AllocationPolicy
+          .newBuilder()
+          .setName(policyName)
+          .setPriority(Int32Value.newBuilder().setValue(1))
+          .build();
+
+      CreateAllocationPolicyRequest request = CreateAllocationPolicyRequest
+          .newBuilder()
+          .setParent(parent)
+          .setAllocationPolicyId(policyId)
+          .setAllocationPolicy(policy)
+          .build();
+
+      RetryingFuture<OperationSnapshot> poll = client.createAllocationPolicyAsync(request)
```

Comment:
Ok so digging into this a bit, it looks like `createAllocationPolicyAsync` already returns an `OperationFuture`. The `getPollingFuture` actually returns a `[RetryingFuture](http://googleapis.github.io/gax-java/latest/apidocs/com/google/api/gax/retrying/RetryingFuture.html)` which is marked as "internal/advanced use only." So I'm still not sure what the PollingFuture does, but I'm pretty sure it's the wrong one. 

I think you probably want something like the following: 
```java
// Send the request to the server to start creating the AllocationPolicy
OperationFuture<AllocationPolicy> futureResponse = client.client.createAllocationPolicyAsync(request);
try {
    // Wait until the AllocationPolicy is created successfully
    AllocationPolicy response = futureResponse.get(1, TimeUnit.MINUTES);
    System.out.println("Allocation Policy created successfully: " + response.getName());
} catch (InterruptedException | TimeoutException e) {
    System.out.println("Unable to receive response from server: " + e);
}
```
I can't quite tell if the `create` api call is actually being performed asynchronously or if it is synchronous and the future represents a constant polling until complete. It is hard to tell from the documentation.

Reply:
All right, I changed these out for just getting the response from the call directly.